src/persist-client/src/fetch.rs

```diff
@@ -635,10 +643,14 @@ where
     D: Semigroup + Codec64 + Send + Sync,
 {
     /// [Self::next] but optionally providing a `K` and `V` for alloc reuse.
+    ///
+    /// When `result_override` is specified, return it instead of decoding data.
+    /// This is used when we know the decoded result will be ignored.
```

Comment:
A bit of a surprising API, but all the alternatives I can imagine are much more invasive, so this feels like the right call to me also.